### PR TITLE
Add support for `BENTLEY_materials_line_style`

### PR DESCRIPTION
**Pull request by j9liu**
## Description

See https://github.com/CesiumGS/glTF/pull/89 for the spec.

I regenerated the classes after deleting the `generated` folders, which seemed to flush out some unused classes (hence the random file edits included in the diff).

## Issue number or link

N/A

## Author checklist

- [X] I have submitted a [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) (only needed once).
- [x] I have done a full self-review of my code.
- [X] I have updated `CHANGES.md` with a short summary of my change (for user-facing changes).
- [X] I have added or updated unit tests to ensure consistent code coverage as necessary.
- [x] I have updated the documentation as necessary.

## Testing plan

No runtime rendering implementation yet, but hopefully the unit tests suffice.

## Reviewer checklist

Thank you for taking the time to review this PR. By approving a PR you are taking as much responsibility for these changes as the author.

As you review, please go through the checklist below:

- [ ] Review and run all parts of the test plan on this branch and verify it matches expectations.
    - [ ] If the issue is a bug please make sure you can reproduce the bug in the main branch and then checkout this branch to make sure it actually solved the issue.
- [ ] Review the code and make sure you do not have any remaining questions or concerns. You should understand the code change and the chosen approach. If you are not confident or have doubts about the code, please do not hesitate to ask questions.
    - [ ] Code should follow the [C++ Style Guide](https://cesium.com/learn/cesium-native/ref-doc/style-guide.html)
- [ ] Review the unit tests and make sure there are no missing tests or edge cases.
- [ ] Review documentation changes and updates to `CHANGES.md` to make sure they accurately cover the work in this PR.
- [ ] Verify that the [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) has been submitted, if needed.

### CODE REVIEW

File: CesiumGltfReader/test/TestGltfReader.cpp [hunk 1]
```diff
@@ -279,6 +281,34 @@ TEST_CASE("Read TriangleWithPaddingInGlbBin") {
...
+  GltfReaderResult result = reader.readGltf(data);
+  REQUIRE(result.model);
+
+  const Model& model = result.model.value();
+  REQUIRE_EQ(model.meshes.size(), 1);
+
+  REQUIRE_EQ(model.meshes[0].primitives.size(), 1);
+
+  const CesiumGltf::MeshPrimitive& primitive = model.meshes[0].primitives[0];
+  CHECK(primitive.hasExtension<ExtensionExtMeshPrimitiveEdgeVisibility>());
```
Comment (timoore) [inline CesiumGltfReader/test/TestGltfReader.cpp:299]: Why is this being checked in this test?

**Review by timoore — COMMENTED** [review]
Thanks for the contribution @j9liu ! It's kind of a shame that we have to take the glTF repo as it is without really tracking its changes in our history, but that's just the way it is. The only comment I have is about the edge visibility test.

File: CesiumGltfReader/test/TestGltfReader.cpp (hunk 1 above)
Comment (j9liu) [reply CesiumGltfReader/test/TestGltfReader.cpp:299]: I mainly put it because the primitive itself is a triangle mesh, and putting a check for `primitive.mode = TRIANGLES` seemed counterintuitive. It's the edges that are affected by `BENTLEY_materials_line_style`.
